File: Learn_Ruby_the_Hard_Way/ex44/ex44c.rb
class Parent
  def altered()
    puts "PARENT altered()"
  end
end

class Child < Parent
  def altered()
    puts "CHILD, BEFORE PARENT altered()"
    super()
    puts "CHILD, AFTER PARENT altered()"
    super()
    puts "CHILD, AFTER PARENT altered() CALLED TWICE"
  end
end

dad = Parent.new()
son = Child.new()

dad.altered()
puts "-" * 10
son.altered()

File: LPtHW/ex44/ex44c.py
class Parent(object):

    def altered(self):
        print "PARENT altered()"

class Child(Parent):

    def altered(self):
        print "CHILD, BEFORE PARENT altered()"
        # call super with args Child and self, 
        # then find from the next class above Child 
        #    that has a method called "altered"
        # and run whatever code the method contains
        super(Child, self).altered()
        # contiune with the code block until whatever.
        print "CHILD, AFTER PARENT altered()"

dad = Parent()
son = Child()

dad.altered()
print "-" * 10
son.altered()
